homeassistant/components/blink/alarm_control_panel.py

```diff
@@ -59,49 +67,43 @@ def __init__(self, data, name: str, sync) -> None:
             name=f"{DOMAIN} {name}",
             manufacturer=DEFAULT_BRAND,
         )
-
-    async def async_update(self) -> None:
-        """Update the state of the device."""
-        if self.data.check_if_ok_to_update():
-            _LOGGER.debug(
-                "Initiating a blink.refresh() from BlinkSyncModule('%s') (%s)",
-                self._name,
-                self.data,
-            )
-            try:
-                await self.data.refresh(force=True)
-                self._attr_available = True
-            except asyncio.TimeoutError:
-                self._attr_available = False
-
-            _LOGGER.info("Updating State of Blink Alarm Control Panel '%s'", self._name)
-
-        self.sync.attributes["network_info"] = self.data.networks
+        self._update_attr()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle coordinator update."""
+        self._update_attr()
+        super()._handle_coordinator_update()
+
+    @callback
+    def _update_attr(self) -> None:
+        """Update attributes."""
+        self.sync.attributes["network_info"] = self.api.networks
         self.sync.attributes["associated_cameras"] = list(self.sync.cameras)
         self.sync.attributes[ATTR_ATTRIBUTION] = DEFAULT_ATTRIBUTION
         self._attr_extra_state_attributes = self.sync.attributes
-
-    @property
-    def state(self) -> StateType:
-        """Return state of alarm."""
-        return STATE_ALARM_ARMED_AWAY if self.sync.arm else STATE_ALARM_DISARMED
+        self._attr_state = (
+            STATE_ALARM_ARMED_AWAY if self.sync.arm else STATE_ALARM_DISARMED
+        )
 
     async def async_alarm_disarm(self, code: str | None = None) -> None:
         """Send disarm command."""
         try:
             await self.sync.async_arm(False)
-            await self.sync.refresh(force=True)
-        except asyncio.TimeoutError:
-            self._attr_available = False
 
+        except asyncio.TimeoutError as er:
+            raise HomeAssistantError("Blink failed to disarm camera") from er
+
+        await self._coordinator.async_refresh()
         self.async_write_ha_state()
```

Comment:
I think this will result in double state writes as when the coordinator receives new data, the `_handle_coordinator_update` is called, which also writes a state.

Reply:
I agree.